### PR TITLE
Add missing API calls for Cloudstack 4.20

### LINKED ISSUE #22: Missing API support
Here's a list of the APIs in 4.16 not yet added to the SDK

```
acquirePodIpAddress
addAnnotation
addCiscoAsa1000vResource
addCiscoVnmcResource
addExternalFirewall
addExternalLoadBalancer
addF5LoadBalancer
addKubernetesSupportedVersion
addSrxFirewall
addUserToProject
addVmwareDc
archiveSnapshot
cancelHostAsDegraded
cloudianIsEnabled
configureF5LoadBalancer
configureSrxFirewall
createKubernetesCluster
createManagementNetworkIpRange
createProjectRole
createSnapshotFromVMSnapshot
declareHostAsDegraded
deleteCiscoAsa1000vResource
deleteCiscoNexusVSM
deleteCiscoVnmcResource
deleteExternalFirewall
deleteExternalLoadBalancer
deleteF5LoadBalancer
deleteKubernetesCluster
deleteKubernetesSupportedVersion
deleteManagementNetworkIpRange
deleteNetscalerControlCenter
deleteProjectRole
deleteResourceIcon
deleteServicePackageOffering
deleteSrxFirewall
deleteUserFromProject
deployNetscalerVpx
destroyVolume
disableCiscoNexusVSM
disableHAForHost
enableCiscoNexusVSM
getDiagnosticsData
getKubernetesClusterConfig
getRouterHealthCheckResults
getSolidFireVolumeAccessGroupIds
getUploadParamsForIso
getUserKeys
importRole
importUnmanagedInstance
importVsphereStoragePolicies
issueCertificate
linkAccountToLdap
listAnnotations
listCaCertificate
listCAProviders
listCiscoAsa1000vResources
listCiscoNexusVSMs
listCiscoVnmcResources
listDetailOptions
listElastistorInterface
listElastistorPool
listElastistorVolume
listExternalFirewalls
listExternalLoadBalancers
listF5LoadBalancerNetworks
listF5LoadBalancers
listHostHAProviders
listHostHAResources
listInfrastructure
listKubernetesClusters
listKubernetesSupportedVersions
listManagementServers
listNetscalerControlCenter
listProjectRoles
listRegisteredServicePackages
listResourceIcon
listSrxFirewallNetworks
listSrxFirewalls
listStoragePoolsMetrics
listUnmanagedInstances
listVmwareDcs
listVolumesMetrics
listVsphereStoragePolicies
listVsphereStoragePolicyCompatiblePools
listZonesMetrics
migrateNetwork
migrateSecondaryStorageData
migrateVPC
moveNetworkAclItem
moveUser
provisionCertificate
recoverVolume
registerNetscalerControlCenter
registerNetscalerServicePackage
releasePodIpAddress
removeAnnotation
removeVmwareDc
revokeCertificate
revokeTemplateDirectDownloadCertificate
runDiagnostics
scaleKubernetesCluster
startKubernetesCluster
startRollingMaintenance
stopKubernetesCluster
stopNetScalerVpx
syncStoragePool
unmanageVirtualMachine
updateAnnotationVisibility
updateImageStore
updateKubernetesSupportedVersion
updatePodManagementNetworkIpRange
updateProjectRole
updateSecurityGroup
updateSiocInfo
updateStorageCapabilities
updateVlanIpRange
updateVmwareDc
upgradeKubernetesCluster
uploadResourceIcon
uploadTemplateDirectDownloadCertificate
```

### PR DESCRIPTION
**Pull request by phsm**
Fixes #22

I've noticed that a lot of API calls are missing from the `cloudstack/` directory.

Thus, I took listApis output from the latest Cloudstack version at the time of writing this, 4.20.0.0, and added all those missing calls to the `layout.go`.

There are 2 commits in the PR:
- One updates listApis to the Cloudstack version 4.20.0. For some reason several API calls were not present in my `listApis` output, namely: 
```
listVmwareDcVms
addVmwareDc
listVmwareDcs
removeVmwareDc
updateVmwareDc
addKubernetesSupportedVersion
createKubernetesCluster
deleteKubernetesCluster
deleteKubernetesSupportedVersion
getKubernetesClusterConfig
listKubernetesClusters
listKubernetesSupportedVersions
scaleKubernetesCluster
startKubernetesCluster
stopKubernetesCluster
updateKubernetesSupportedVersion
upgradeKubernetesCluster
addVirtualMachinesToKubernetesCluster
removeVirtualMachinesFromKubernetesCluster
createServiceInstance
cloudianSsoLogin
```
I have ported them from listApis.json from the commit d4d7f16, except `cloudianSsoLogin` which I could not find anywhere. Therefore, I've removed this call from `layout.go`.

This commit also includes changes to `layout.go` that introduces new API calls, and also moves some Netscaler-related API calls to the new Service `NetscalerService`.

- The second commit contains all the files that `make all` generated

There were no errors/warnings during the make all.

### CODE REVIEW

**Review by Pearl1594 — APPROVED** [review]
code built successfully. code LGTM

**Comment by DaanHoogland** [comment]
@phsm I merged some PRs and now you are stuck with a lot of conflicts. So for I can see these are not complex, just a lot. Can you have a look, please?

**Comment by phsm** [comment]
I see you've merged the previous commits implementing the slices of API calls that were emerged at different CS versions. 

This may as well render my PR obsolete.

Later I will check the build against CS 4.20  `listApis` output (thankfully the code generator emits warnings when it encounters an unknown API call), and will adjust my PR to have what is missing.

**Comment by phsm** [comment]
@DaanHoogland updated both the code, and the PR header/description.

File: generate/layout.go [hunk 1]
```diff
@@ -790,16 +864,92 @@ var layout = apiInfo{
 	"RegistrationService": {
 		"registerOauthProvider",
 	},
-	"ShutdownService": {
-		"readyForShutdown",
-	},
```
Comment (Pearl1594) [inline generate/layout.go:795]: any specific reason to remove this? @phsm as the generate.go file has a reference to this api, causing issues:

```
test/ShutdownService_test.go:42:15: client.Shutdown undefined (type *cloudstack.CloudStackClient has no field or method Shutdown)
test/ShutdownService_test.go:43:20: client.Shutdown undefined (type *cloudstack.CloudStackClient has no field or method Shutdown)
```

**Comment by Pearl1594** [comment]
For some reason, the tests seem to fail on this PR - @phsm can you please help us with this:
I needed to make the following changes to make it work - I am not sure why the 1st two changes are required, as it isn't required upstream:

```
diff --git a/Makefile b/Makefile
index 015bba8..33ad3ea 100644
--- a/Makefile
+++ b/Makefile
@@ -27,4 +27,4 @@ test:
 
 MOCKGEN := mockgen
 mockgen: ## Download conversion-gen locally if necessary.
-       go get github.com/golang/mock/mockgen; go install github.com/golang/mock/mockgen;
+       go get go.uber.org/mock/mockgen; go install go.uber.org/mock/mockgen;
```
examples/mock_test.go file
```

diff --git a/examples/mock_test.go b/examples/mock_test.go
index b965c3a..da50adb 100644
--- a/examples/mock_test.go
+++ b/examples/mock_test.go
@@ -23,7 +23,7 @@ import (
        "testing"
 
        "github.com/apache/cloudstack-go/v2/cloudstack"
-       "github.com/golang/mock/gomock"
+       "go.uber.org/mock/gomock"
 )
 
 func Test_Mock(t *testing.T) {

```

layout.go
```
git diff generate/layout.go
diff --git a/generate/layout.go b/generate/layout.go
index d399ba9..2271abb 100644
--- a/generate/layout.go
+++ b/generate/layout.go
@@ -864,6 +864,9 @@ var layout = apiInfo{
        "RegistrationService": {
                "registerOauthProvider",
        },
+       "ShutdownService": {
+               "readyForShutdown",
+       },
        "DiagnosticsService": {
                "getDiagnosticsData",
                "runDiagnostics",
```

File: generate/layout.go (hunk 1 above)
Comment (phsm) [reply generate/layout.go:795]: The reason was to make the layout structure to follow the structure from here: https://cloudstack.apache.org/api/apidocs-4.20/

The "readyForShutdown" API call looks to be a part of ManagementService now.

**Comment by phsm** [comment]
@Pearl1594  This was to tidy up an abandoned https://github.com/golang/mock/ package.
The authors there suggest to switch to a maintained fork instead.

I replaced the rest of the old package references with `go.uber.org/mock/gomock`, lets see if the tests will work now.

**Comment by Pearl1594** [comment]
Thanks @phsm - tests now run successfully! Thank you :+1:

**Review by DaanHoogland — APPROVED** [review]

**Pull request merged**